x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/gather_referenced_exceptions.ts

```diff
@@ -11,29 +11,28 @@ import { getAllListTypes } from '@kbn/lists-plugin/server/services/exception_lis
 import type { RuleToImport } from '../../../../../../common/detection_engine/rule_management';
 
 /**
- * Helper that takes rules, goes through their referenced exception lists and
- * searches for them, returning an object with all those found, using list_id as keys
- * @param rules {array}
- * @param savedObjectsClient {object}
- * @returns {Promise} an object with all referenced lists found, using list_id as keys
+ * splitting out the parsing of the lists from the fetching
+ * for easier and more compartmentalized testing
+ * @param rules Array<RuleToImport | Error>
+ * @returns [ExceptionListQueryInfo[], ExceptionListQueryInfo[]]
  */
-export const getReferencedExceptionLists = async ({
-  rules,
-  savedObjectsClient,
-}: {
-  rules: Array<RuleToImport | Error>;
-  savedObjectsClient: SavedObjectsClientContract;
-}): Promise<Record<string, ExceptionListSchema>> => {
-  const [lists] = rules.reduce<ListArray[]>((acc, rule) => {
-    if (!(rule instanceof Error) && rule.exceptions_list != null) {
-      return [...acc, rule.exceptions_list];
```

Comment:
Looking at the original code [here](https://github.com/elastic/kibana/blame/8.5/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils/gather_referenced_exceptions.ts#L27-L29), we see that there is a bug that the unit testing code missed due to reliance on the mocked results of the saved objects client. 

The bug is as-follows: If importing multiple rules where the first rule's `exception_list` was an empty array, this code would result in the removal of all of the other rules' references to any other exception lists. This is detrimental to the user experience and was brought to our attention via [this community slack thread.](https://elasticstack.slack.com/archives/C016E72DWDS/p1665021134894019)